system/doc/design_principles/sup_princ.md

```diff
@@ -146,8 +145,9 @@ flowchart TD
 
 ### one_for_all
 
-If a child process terminates, all other child processes are terminated, and
-then all child processes, including the terminated one, are restarted.
+If a child process terminates, all remaining child processes are
+terminated. Subsequently, all child processes, including the
+terminated one, are restarted.
```

Comment:
Maybe mention termination and restart order?

Reply:
It is mentioned in the beginning under Supervision Principles:

> The child processes are started in the order specified by this list, and are terminated in the reverse order.

I am not sure that it is necessary to repeat that.